lib/SILOptimizer/Analysis/PassManagerVerifierAnalysis.cpp

```diff
@@ -0,0 +1,100 @@
+//===--- PassManagerVerifierAnalysis.cpp ----------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "sil-passmanager-verifier-analysis"
+#include "swift/SILOptimizer/Analysis/PassManagerVerifierAnalysis.h"
+#include "swift/SIL/SILModule.h"
+#include "llvm/Support/CommandLine.h"
+
+static llvm::cl::opt<bool>
+    EnableVerifier("enable-sil-passmanager-verifier-analysis",
+                   llvm::cl::desc("Enable verification of the passmanagers "
+                                  "function notification infrastructure"),
+                   llvm::cl::init(false));
+
+using namespace swift;
+
+PassManagerVerifierAnalysis::PassManagerVerifierAnalysis(SILModule *mod)
+    : SILAnalysis(SILAnalysisKind::PassManagerVerifier), mod(*mod) {
+#ifndef NDEBUG
+  if (!EnableVerifier)
+    return;
+  for (auto &fn : *mod) {
+    DEBUG(llvm::dbgs() << "PMVerifierAnalysis. Add: " << fn.getName() << '\n');
+    liveFunctions.insert(&fn);
+  }
+#endif
+}
+
+/// Validate that the analysis is able to look up all functions and that those
+/// functions are live.
```

Comment:
I don't understand this comment at all. I think we just don't track invalidation.